### ISSUE TITLE
Add XGBoost probability calibration so probability_up reflects true frequencies

### ISSUE THREAD

**Issue by sjg6590** [sjg6590/stock-decision-assistant-ml#16]
## Problem

`XGBClassifier.predict_proba()` does not output well-calibrated probabilities — the raw sigmoid outputs are systematically over-confident (spiky toward 0 and 1) compared to true empirical frequencies. These raw values flow directly into `src/ml/predict.py` as `probability_up` and are averaged into `ml_confidence`, which is then used by `src/signals/fuse.py` to gate BUY alerts via `sentiment_confidence_min`. A `probability_up` of 0.70 does not mean "this stock goes up 70% of the time at this threshold" — it is an uncalibrated score, making both the gating logic and any user-facing confidence display misleading.

## Goal

Wrap each `XGBClassifier` with isotonic-regression calibration fitted on the validation set so that `probability_up` outputs approximate true empirical probabilities on held-out data.

## User story

As a user reading alert confidence values, I want `probability_up = 0.65` to mean the model is right ~65% of the time at that threshold so that I can trust confidence numbers when deciding whether to act on a signal.

### Phase A — Calibrate on the validation set

After fitting each horizon's classifier in `train.py`, wrap it with `sklearn.calibration.CalibratedClassifierCV`:

```python
from sklearn.calibration import CalibratedClassifierCV

# After clf.fit(x_train, y_train, ...)
calibrated_clf = CalibratedClassifierCV(clf, method="isotonic", cv="prefit")
calibrated_clf.fit(x_val, y_val)
model_payload["classifiers"][hz] = calibrated_clf
```

`cv="prefit"` tells sklearn the base estimator is already fitted; it only learns the isotonic mapping on `(x_val, y_val)`. Store the calibrated classifier in the payload. Inference in `predict.py` is unchanged — `predict_proba` on a `CalibratedClassifierCV` returns calibrated probabilities.

Apply to both the seed_only/rolling_holdout path and the purged_cv path (use the last CV fold's val set for calibration).

### Phase B — Calibration quality check

Add a calibration metric to the holdout evaluation: compute Expected Calibration Error (ECE) in 10 bins and log it alongside Sharpe and accuracy. Add a `max_ece` promotion gate (optional, off by default).

### Phase C — Tests

Add `tests/test_ml_calibration.py`:
- Train a toy classifier, calibrate on val, assert ECE on holdout is lower than uncalibrated ECE
- Assert `predict_proba` output shape is unchanged after calibration wrap

## Acceptance criteria

- [ ] Each horizon's classifier in the saved payload is a `CalibratedClassifierCV` wrapping the raw `XGBClassifier`
- [ ] `predict.py` inference is unchanged (no code changes required there)
- [ ] Calibration is applied on the val set, never on the holdout
- [ ] ECE is logged at DEBUG per horizon per attempt
- [ ] `pytest tests/test_ml_calibration.py` passes
- [ ] Existing `pytest` suite passes

## Technical notes

- `CalibratedClassifierCV(clf, method="isotonic", cv="prefit")` requires `scikit-learn >= 0.24`; already in the dependency tree
- Isotonic regression requires ~50+ calibration samples per class to be reliable; log a WARNING if `len(x_val) < 100`
- The `feature_names_in_` attribute used by `model_features_stale()` lives on the inner `clf`; ensure `resolve_model_features()` in `features.py` still finds it via `calibrated_clf.estimator`
- Platt scaling (`method="sigmoid"`) is an alternative but isotonic is more flexible for large val sets

## Depends on

None (can be implemented independently, but benefits from #09 being fixed first so val data is clean)

## Out of scope (v1)

- Temperature scaling (simpler but less flexible)
- Calibrating the `XGBRegressor` expected-return head
- Adding a calibration plot to the debug visualizations

## Part of epic

ML quality improvements (2026-05)
